notion-server/src/ethereum/mod.rs

```diff
@@ -279,6 +311,39 @@ pub mod args {
 
     // Return values AND/OR sub-components of incoming argument values
 
+    #[derive(Debug, Default)]
+    pub struct BeaconBlock {
+        pub cross_links: HashMap<u32, CrossLink>,
+    }
+    impl From<&super::BeaconBlock> for BeaconBlock {
+        fn from(beacon_block: &super::BeaconBlock) -> Self {
+            let mut cross_links: HashMap<u32, CrossLink> = HashMap::new();
+            for (k, v) in &beacon_block.cross_links {
```

Comment:
Absolutely nothing wrong with how this is currently written, but I'd like to mention that, if you prefer a functional style, you can use [`Iterator::collect`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) with `HashMap`.

Reply:
Totally fair, I'll consider switching to that, may be more concise, and think I use it in another spot.  No promises since this approach I think has advantages re: readability but I'll definitely compare the two.  Thanks for heads up.